### PR TITLE
Fix initialization of BatchNorm

### PR DESCRIPTION
**Pull request by gphilippee**
# Problem

Difficulty to reproduce Fig 2.d. (sequence of pMNIST) and Fig 4.b. (MNIST-FMNIST). In recent Pytorch versions, the elements of $\gamma$ are set to 1 (cf [doc](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html#torch.nn.BatchNorm1d)). In the Pytorch version of the paper (1.1.0), elements of $\gamma$ were initialized uniformly (cf [link](https://pytorch.org/docs/1.1.0/nn.html#batchnorm1d)).

# Fix

Initialize BatchNorm weights uniformly. Currently I have only tested it with the BNN. More testing needs to be done with the BCNN and other datasets.

# Comment

I'm not sure if the PR should be merged but it will give a lead to anyone looking to reproduce the results with recent Pytorch versions.

### CODE REVIEW

**Comment by Laborieux-Axel** [comment]
Hi Guillaume,

Thanks for looking into this. This is a recurrent issue raised by people using the code with newer PyTorch versions. 

I have also heard about a fix consisting in setting the `track_running_stats` argument of BN to `False`, which seems unrelated to the fix proposed here, so maybe the root cause is in the way PyTorch updates the running stats? 

I would be happy to merge a fix if the exact reason is pinpointed. Unfortunately, I don't have time to look into this at the moment...
